edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/WorkloadClientVersioned.cs

```diff
@@ -55,13 +59,27 @@ protected internal async Task<T> Execute<T>(Func<Task<T>> func, string operation
             try
             {
                 Events.ExecutingOperation(operation, this.WorkloadUri.ToString());
-                T result = await ExecuteWithRetry(func, r => Events.RetryingOperation(operation, this.WorkloadUri.ToString(), r), this.transientErrorDetectionStrategy)
+                T result = await ExecuteWithRetry(
+                    func,
+                    r => Events.RetryingOperation(operation, this.WorkloadUri.ToString(), r),
+                    this.transientErrorDetectionStrategy)
                     .TimeoutAfter(this.operationTimeout);
                 Events.SuccessfullyExecutedOperation(operation, this.WorkloadUri.ToString());
+                this.failedSocketCount = 0;
                 return result;
             }
             catch (Exception ex)
             {
+                if (this.IsWorkloadSocketStale(ex))
+                {
+                    this.failedSocketCount++;
+                    Events.ErrorExecutingOperation(ex, operation, this.WorkloadUri.ToString());
+                    if (this.failedSocketCount >= MaxRetryCount)
+                    {
+                        Environment.Exit(0);
+                    }
+                }
+
                 this.HandleException(ex, operation);
                 Events.SuccessfullyExecutedOperation(operation, this.WorkloadUri.ToString());
```

Comment:
Remove it to fix this legacy bug and call Events.ErrorExecutingOperation first right after caught instead of line76 same as what did in Module management client.